node/src/components/storage.rs

```diff
@@ -307,6 +308,44 @@ impl Storage {
         drop(cursor);
         drop(block_txn);
 
+        info!("Checking block body db");
+        let body_txn = env.begin_ro_txn()?;
+        let mut cursor = body_txn.open_ro_cursor(block_body_db)?;
+
+        for (raw_key, raw_val) in cursor.iter() {
+            let body: BlockBody = lmdb_ext::deserialize(raw_val)?;
+            assert_eq!(
+                raw_key,
+                body.hash().as_ref(),
+                "found corrupt block body in database"
+            );
+        }
+        info!("block body db check complete");
+        drop(cursor);
+        drop(body_txn);
```

Comment:
I am still seeing `drop`s called explicitly here for some reason. I thought you had removed them.

Reply:
It says `outdated` at the top, perhaps from an earlier commit??

Reply:
This is done elsewhere instead of just calling [lmdb::Transaction::abort][1] directly.

[1]: https://docs.rs/lmdb-rkv/0.14.0/lmdb/trait.Transaction.html#method.abort

Reply:
`drop` wouldn't abort the txn though so they're different things.